nncf/experimental/post_training/statistics/aggregator.py

```diff
@@ -32,19 +31,21 @@ class StatisticsAggregator(ABC):
     Base class for statistics collection.
     """
 
-    def __init__(self, engine: Engine, dataset: Dataset):
+    def __init__(self, engine: Engine, dataset: Dataset, model_transformer: StaticModelTransformerBase):
         self.engine = engine
         self.dataset = dataset
         self.is_calculate_metric = False
         self.max_number_samples = 0
         self.statistic_points = StatisticPointsContainer()
+        self.model_transformer = model_transformer
 
     def collect_statistics(self, model: ModelType) -> None:
         """
         Collects statistics for registered StatisticPoints.
         The statistics are stored in self.statistic_points.
         """
-        self.engine.set_model(self._prepare_model_for_statistics_collection(model))
+        self.model_transformer.set_model(model)
```

Comment:
We should collect statistics in the case of different algorithms that wouldn't know about others. Checking the length will break this logic.

Reply:
I mean it doesn't make any sense to run this function if `self.statistic_points` doesn't have statistics points.
I suggest to check whether there is `self.statistic_points`. If not raise an Exception

Reply:
Done

Reply:
@vshampor @alexsu52 could you help us with this moment? 
We have a quantization pipeline: 
We register quantization target points, which describe from which activation tensor we should collect the statistics. So in most cases there are models that have activation quantizers, but in the synthetic cases, or maybe some specific models could have no activation quantizers. Because these cases are not typical for us, I suggest to add a check that we have registered activation quantizers before collecting statistics. 
1) It saves time as we don't need to do extra steps like preparing model for collecting statistics and so on.
2) If there is no quantizes add warning message that there is a problem in quantization algorithm or this is the specific model in which we didn't find any position for quantizer.
So the question is - Is it a good idea to add such check?

Reply:
That's a pretty strange situation that you described. I would focus on more relevant problems.